### PR TITLE
✨ VAUS ball bounce

### PR DESCRIPTION
**Pull request by maelehrhard**

### CODE REVIEW

File: arkanoid.c [hunk 1]
```diff
@@ -46,31 +64,47 @@ void draw() {
...
+            vaus.moving_direction == LEFT && ball_movement.x > 0 ||
+            vaus.moving_direction == RIGHT && ball_movement.x > 0;
+        const bool is_moving_same =
+            vaus.moving_direction == LEFT && ball_movement.x < 0 ||
+            vaus.moving_direction == RIGHT && ball_movement.x < 0;
+        if (is_moving_opposite) {
+            ball.direction += BALL_EFFECT;
+        } else if (is_moving_same) {
+            ball.direction -= BALL_EFFECT;
+        }
```
Comment (axelmunch) [inline arkanoid.c]: After some testing, the ball_movement has no influence on bounce effect.
The tests can only be vaus.moving_direction == LEFT or vaus.moving_direction == Right

File: arkanoid.c [hunk 2]
```diff
@@ -46,31 +64,47 @@ void draw() {
...
-    draw_integer(win_surf, (int) get_current_fps(), 10 + fps_text_width, 40);
+    if (collide_with_vaus_x || collide_with_vaus_y) {
+        const bool is_moving_opposite =
+            vaus.moving_direction == LEFT && ball_movement.x > 0 ||
+            vaus.moving_direction == RIGHT && ball_movement.x > 0;
+        const bool is_moving_same =
+            vaus.moving_direction == LEFT && ball_movement.x < 0 ||
+            vaus.moving_direction == RIGHT && ball_movement.x < 0;
+        if (is_moving_opposite) {
+            ball.direction += BALL_EFFECT;
```
Comment (axelmunch) [inline arkanoid.c]: Maybe use fmod(, 360)

File: arkanoid.c [hunk 3]
```diff
@@ -46,31 +64,47 @@ void draw() {
...
+        } else if (is_moving_same) {
+            ball.direction -= BALL_EFFECT;
```
Comment (axelmunch) [inline arkanoid.c]: Maybe use fmod(, 360)

File: arkanoid.c (hunk 1 above)
Comment (maelehrhard) [reply arkanoid.c]: fixed

File: arkanoid.c (hunk 2 above)
Comment (maelehrhard) [reply arkanoid.c]: fixed

File: arkanoid.c (hunk 3 above)
Comment (maelehrhard) [reply arkanoid.c]: fixed

**Review by axelmunch — APPROVED** [review]

**Pull request merged**